lib/views/qr_reader_view_mixin.dart

```diff
@@ -29,44 +31,48 @@ mixin QrReaderViewMixin<T extends StatefulWidget> on State<T> {
       ..addListener(_upState)
       ..addStatusListener((state) {
         if (state == AnimationStatus.completed) {
-          timer = Timer(Duration(seconds: 1), () {
-            animationController.reverse(from: 1.0);
+          timer = Timer(const Duration(seconds: 1), () {
+            animationController.reverse(from: 1);
           });
         } else if (state == AnimationStatus.dismissed) {
-          timer = Timer(Duration(seconds: 1), () {
-            animationController.forward(from: 0.0);
+          timer = Timer(const Duration(seconds: 1), () {
+            animationController.forward(from: 0);
           });
         }
-      });
-    animationController.forward(from: 0.0);
+      })
+      ..forward(from: 0);
   }
 
   void clearAnimation() {
-    animationController
-      ..stop()
-      ..dispose();
+    animationController.dispose();
     timer?.cancel();
   }
 
   void _upState() {
-    setState(() {});
+    if (mounted) setState(() {});
   }
 
   Future<void> onCreateController(QrReaderViewController qrReaderViewController) async {
     controller = qrReaderViewController;
-    controller.startCamera(_onQrBack);
+    await controller.startCamera(_onQrBack);
   }
 
-  Future _onQrBack(data, _, rawData) async {
+  Future<void> _onQrBack(String? data, List<Offset> _, String? rawData) async {
     if (isScan == true) return;
     isScan = true;
-    await onScan(data, rawData);
-    await Future.delayed(Duration(seconds: 2));
-    isScan = false;
+    stopScan();
+    await onScan(data, rawData).then(
+      (value) async {
+        if (!value) {
+          isScan = false;
+          await Future<void>.delayed(const Duration(seconds: 1));
+          await controller.startCamera(_onQrBack);
+        }
+      },
+    );
```

Comment:
Is this file where you actually added the fix?

Reply:
The error occurs when `stopCamera` method is not yet completed while the `dispose` method is being executed, the error will be fixed if i call the `stopCamera` method when it reads a qr code, but if the `dispose` method is called without reading the qr code, the same error will occur again, because `stopCamera` is a future and I can't call it in the `dispose` method, and according to researches this is a java bug and i will continue my researches.

The error that occur after scanning will be solved by this change.

https://github.com/encointer/flutter_qr_reader/blob/9103cca12f70db5f0936cdcfb0e94b6cc01aa1e9/lib/views/qr_reader_view_mixin.dart#L60-L73